src/base/path.cpp

```diff
@@ -127,7 +133,13 @@ Path Path::parentPath() const
     if (slashIndex == 0) // *nix absolute path
         return (m_pathStr.size() == 1) ? Path() : createUnchecked(u"/"_qs);
 
-    return createUnchecked(m_pathStr.left(slashIndex));
+    const QString parent = m_pathStr.left(slashIndex);
+#ifdef Q_OS_WIN
+    // should be `c:/` instead of `c:`
+    if ((parent.back() == u':') && !parent.contains(u'/'))
```

Comment:
I would perform any checking before extracting substring, e.g.:
```c++
#ifdef Q_OS_WIN
if ((slashIndex == 2) && (m_pathStr[1] == u':'))
    return createUnchecked(m_pathStr.left(slashIndex + 1));
#endif
return createUnchecked(m_pathStr.left(slashIndex));
```

Reply:
>if ((slashIndex == 2) && (m_pathStr[1] == u':'))

I don't think this is robust. Is the name of disk drives limited to a single alphabet?

Reply:
> Is the name of disk drives limited to a single alphabet?

It isn't "name of disk drive" but "drive **letter**".

Reply:
>It isn't "name of disk drive" but "drive letter".

I meant the same thing. Anyway, what happens when there are more than 26 drives?

Reply:
Off topic, does `Path` support UNC path?

Reply:
> what happens when there are more than 26 drives? or when all single alphabet are exhausted?

It doesn't concern us. The concept is that this is exactly the "drive **letter**".


Some systems allow you to assign a non-alphabetic character as a drive letter, but in any case it is the single character/letter. In Windows, you can connect disks via NTFS mount points (similar to *nix).

Reply:
Any way this thread is addressed in the last push.